### PR TITLE
feat: improve MecanumDrive implementation, add autonomous command

### LINKED ISSUE #13: Implement and Improve autonomous Commands for all subsystems.

### PR DESCRIPTION
**Pull request by Trip-kun**
## Summary by CodeRabbit

- **New Features**
	- Introduced new command classes for controlling intake mechanisms and spinner subsystems in autonomous mode.
	- Added functionality for generating robot poses based on camera data using PhotonVision and AprilTags.
	- Enhanced odometry calculations with the `MecanumDriveOdometryWrapper` class for improved pose estimation.
	- Updated `LimitedSpooler` class with a `getSpeed` method to retrieve the current spool speed.

### CODE REVIEW

**Comment by Trip-kun** [comment]
@coderabbitai review

**Comment by Trip-kun** [comment]
@coderabbitai review

**Comment by Trip-kun** [comment]
@coderabbitai review

File: src/main/java/frc/robot/hardware/encoder/NEOEncoder.java [hunk 1]
```diff
@@ -1,15 +1,16 @@
 package frc.robot.hardware.encoder;
 
+import com.revrobotics.*;
```
[the remark this answers is not in the document]
Comment (Baconing) [reply src/main/java/frc/robot/hardware/encoder/NEOEncoder.java:3]: ^

**Review by Baconing — APPROVED** [review]
a lot of odometry stuff i don't know because im not a NERD like you but lgtm, if the bot explodes apart we blame you :)

**Pull request merged**